i2c: RAM test improvements

teburd opened the pull request:
Better support i2c ram tests by signaling that a driver supports RTIO (HAS_I2C_RTIO in Kconfig), and supporting power management in the test case.

Commit:
i2c: Add Kconfig HAS_I2C_RTIO

This config should be selected by any driver that supports the RTIO
interface and can then be checked on by things like tests or drivers
that subsequently require the feature exist.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

Commit:
i2c: sam twihs driver supports rtio

select HAS_I2C_RTIO when sam twihs is selected

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

Commit:
i2c: RAM test updates to support power management

It's important we test the power management flows when testing i2c

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

ubieda commented on drivers/i2c/Kconfig.sam_twihs:
```diff
@@ -7,5 +7,6 @@ config I2C_SAM_TWIHS
 	bool "Atmel SAM (TWIHS) I2C driver"
 	default y
 	depends on DT_HAS_ATMEL_SAM_I2C_TWIHS_ENABLED
+	select HAS_I2C_RTIO
```
Do we need this as well for `I2C_MCUX_LPI2C`?

``` code
config I2C_MCUX_LPI2C
	bool "MCUX LPI2C driver"
	default y
	select HAS_I2C_RTIO
	...
```

teburd replied:
Yes we do, thank you

Commit:
i2c: Fix filtering on alias for i2c-ram

Was previously looking for i2c_ram which is incorrect, the alias was
always i2c-ram.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

hakehuang commented:
@teburd  one enhancement on this, you can move the mimxrt1010_evk.overlay to app.overlay, in this way, all arudio compatible boards can use your apps without code change

Commit:
i2c: LPI2C selects HAS_I2C_RTIO

LPI2C from NXP supports the RTIO interface and therefore now selects the
HAS_I2C_RTIO Kconfig.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

hakehuang commented on tests/drivers/i2c/i2c_ram/testcase.yaml:
```diff
@@ -3,13 +3,27 @@ common:
   tags:
     - drivers
     - i2c
```
as arduino header is required by default, so need add 
```
depends_on: 
    - arduino_i2c
```

teburd replied:
perhaps I simply need to assign this test a fixture or something? It doesn't truly depend on arduino_i2c it depends on i2c-ram existing which I suppose is a test fixture of sorts?

hakehuang replied:
yes, you can do that way, and to make the cse more adaptive to new platforms, we can reuse the arduino_i2c as a sub dependency, so that all boards with arduino_i2c will enabled by default with the test fixture installed. and this can also applys to other common header definition, one step further can save a lot.

teburd replied:
added a fixture, i2c_ram

teburd commented:
@hakehuang I'm giving up on this app.overlay for now, if you have some ideas on how to resolve it please do send a PR but this change is holding up the rest with CI failures
